shenyu-common/src/main/java/org/apache/shenyu/common/timer/WheelTimerFactory.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shenyu.common.timer;
+
+/**
+ * WheelTimerFactory .
+ * shared wheel time.
+ */
+public class WheelTimerFactory {
+    
+    private static final String NAME = "shared_wheel_timer";
+    
+    private static final TimerSharedRef SHARED_TIMER = new TimerSharedRef();
+    
+    /**
+     * Gets wheel timer.
+     *
+     * @return the wheel timer
+     */
+    public static Timer getSharedTimer() {
+        return SHARED_TIMER.getRef();
+    }
+    
+    /**
+     * New wheel timer hashed wheel timer.
+     *
+     * @return the hashed wheel timer
+     */
+    public static Timer newWheelTimer() {
+        return new HierarchicalWheelTimer(NAME);
+    }
+    
+    private abstract static class Shared<T> {
```

Comment:
I think a simple shared supplier can be implemented.

`public class ShardedSupplier<T> implements Supplier<T> {

    private final Supplier<T> delegate;

    private volatile boolean initialized;
    private T value;

    public ShardedSupplier(Supplier<T> delegate) {
        this.delegate = delegate;
    }

    @Override
    public T get() {
        if (!initialized) {
            synchronized (this) {
                if (!initialized) {
                    T t = this.delegate.get();
                    this.value = t;
                    this.initialized = true;
                    return t;
                }
            }
        }
        return this.value;
    }

    public static <T> ShardedSupplier<T> getShardedSupplier(Supplier<T> delegate) {
        return new ShardedSupplier<>(delegate);
    }
}`